lib/wallet/http.js

```diff
@@ -811,7 +811,10 @@ class HTTP extends Server {
       const network = this.network;
       const ns = await req.wallet.getNameStateByName(name);
 
-      res.json(200, ns.getJSON(height, network));
+      if (!ns)
+        return res.json(404);
```

Comment:
@pinheadmz the code base typically calls `res.json()` then calls `return`.

Reply:
that's fine -- the error is `ns.getJSON` when `ns` is `null`

Reply:
or do you mean you prefer an if/else instead of a return?

Reply:
```js
if (!ns) {
  res.json(404);
  return;
}
```

Reply:
hm, I think this is the convention though:

https://github.com/pinheadmz/hsd/blob/2ef54a503db4494270a190eed8910d8bc531d8ca/lib/wallet/http.js#L1142

although, I think I should change the error code to `400` not `404`

Reply:
I guess there isn't a convention lol: https://github.com/handshake-org/hsd/blob/2ef54a503db4494270a190eed8910d8bc531d8ca/lib/wallet/http.js#L218

Reply:
> although, I think I should change the error code to 400 not 404

Agreed.

Reply:
ha! ok how about I just change the error code to `400` and we 🍻

Reply:
Discussed out of band. We're going to leave this as-is for now.